### ISSUE TITLE
Build errors 

### ISSUE THREAD

**Issue by shujaatak** [ThomArmax/arduino-qt-creator-plugin#2]
I am trying to build this Arduino plugin for QtCreator-4.12.0 but I get so many errors. I think you have targeted this Plugin for an older version of QtCreator. 
Errors are like this:

`F:\QtCreator_Arduino\Workspace\arduino-qt-creator-plugin-master\src\toolchain\arduinotoolchain.h:80: error: C2039: 'FileName': is not a member of 'Utils'`

**Comment by shujaatak** [comment]
Can you please mention your QtCreator version and Kit you are using for building this plugin?

**Comment by ThomArmax** [comment]
If I well remember, I used QtCreator 4.9. I'll check tomorrow

**Comment by shujaatak** [comment]
OK, so I will try to test this repository tomorrow.

**Comment by shujaatak** [comment]
Does completion works for Arduino with this plugin?

**Comment by ThomArmax** [comment]
QtCreator has a great auto completion capabilities

**Comment by ThomArmax** [comment]
I have checked, I build the plugin against QtCreator git tag `v4.5.2` sha `3c21c0d80b52eb43483308eccc663279e9f369a4`

**Comment by shujaatak** [comment]
Going to test it now.

**Comment by shujaatak** [comment]
I downloaded the QtCreator source and binary from here: http://download.qt.io/official_releases/qtcreator/4.5/4.5.2/

After installing QtCreator-4.5.2, I checked the version of Qt with which it was built using the QtCreator about page. I noticed that it is based on Qt-5.10.1. As on Windows, the plugins for QtCreator should be built using the same Qt version on which QtCreator is based so I searched Qt-5.10.1 in the http://download.qt.io/ but I could not find it. So I am unable to test this repository as it is based on an old unsupported Qt Version.

It would have been a lot helpful if this repository had targeted a new QtCreator version with LTS Qt version.

**Comment by ThomArmax** [comment]
Actually, you don't have to install this qt-creator version, but download its sources, compile them with whatever Qt version, then compile the plugin.

EDIT : Personnaly, I have built qt-creator 4.5.2 with Qt 5.9.1 64 bits

**Comment by shujaatak** [comment]
Going to do the same.

**Comment by ThomArmax** [comment]
**PS** : I have not tried on Windows, only Linux (KUbuntu 18.04 64bits)

**Comment by shujaatak** [comment]
It's OK, If I faced some minor issues, I would try to resolve them.

**Comment by shujaatak** [comment]
Built both QtCreator and the plugin successfully after adding missing dependencies like this; 

```
QTC_PLUGIN_DEPENDS += \
    coreplugin \
    projectexplorer \
    qtsupport \
    debugger
```

QtCreator shows the Arduino plugin BUT during new project creation, whenever I choose **Arduino Project** as a template the QtCreator crashes. Similarly whenever I click on **Devices** in the Options, the QtCreator crashes.

**Comment by ThomArmax** [comment]
I have a lot of local modifications not pushed on Github yet, since I was trying to implement the Arduino project and toolchain integration.
There is nothing functional yet, this is the very begining, and it will require a lot of work to make a clean integration.
Some other solutions already exists like using Visual Studio Code with Platform IO.
If you want, I can push all my local modifications into a dedicated branch, without any guaranty (not sure it build)

**Comment by shujaatak** [comment]
Yes it would be a lot helpful if you push your local modifications into a dedicated branch. Just like you, I am interested in programming Arduino in QtCreator.

**Comment by shujaatak** [comment]
I am interested in only programming and building Arduino projects in QtCreator, I don't care about Serial-Monitor, Uploading etc.

**Comment by ThomArmax** [comment]
I have pushed the `ft/arduino-kit-integration` branch

**Comment by shujaatak** [comment]
Same behavior, it crashes the QtCreator, I don't know if it's due to the OS difference.

**Comment by ThomArmax** [comment]
I'll check this weekend on my side.

To be sure, if I well remember, you have to:

1. build qt-creator
2. set `IDE_SOURCE_TREE` in `src/src.pro` to the path of qt-creator sources (or pass it in qmake comand line args)
3. set `IDE_BUILD_TREE`  in `src/src.pro` to the path of qt-creator build folder (or pass it in qmake comand line args)
4. run the qt-creator you built in step `1`

**Comment by shujaatak** [comment]
I have followed the same steps, I built Qt-Creator, set the **IDE_SOURCE_TREE** and **IDE_BUILD_TREE** correctly, then successfully **built and deployed the plugin** and then I started the Qt-Creator which I had built(in which the plugin got deployed), **but** as soon as I click on **choose** button while **Arduino Project** selected as a template, the Qt-Creator crash!

By the way, **I am so happy** that you would check this on your side and hopefully you would be able to resolve any issues and thus **arduino-qt-creator-plugin** would become a **cross-platform plugin**!

**Comment by ThomArmax** [comment]
Hi, I have checked on Linux.

State of the art

1. **New Project -> Arduino -> Arduino Project**
    `Arduino has not been configured`
    I have not made the KIT matching between arduino project and avr-gcc toolchain
2. **New Project -> Arduino -> AVR C Project**
    Lead to soft assert `SOFT ASSERT: "wizard" in file ../../../../qt-creator/src/plugins/coreplugin/basefilewizardfactory.cpp, line 96`
    Does not crash in debug, may crash on Windows
    This is because I have not fully implemented the wizard for this type of project.
3. **New Project -> AVR -> C/C++ AVR**
    `Arduino has not been configured` (I have not made the KIT matching between arduino project and avr-gcc toolchain)
    You can chose another Kit (but should not because not compatible)
    Default main project doesn't build since wrong toolchain, maybe also something wrong in `main.cpp`

There is nothing functional yet, only a very small base
